parsers/CL_SING.py

```diff
@@ -203,6 +203,16 @@ def fetch_production(zone_key='CL-SING', session=None, target_datetime=None, log
         }
         production_mix_by_hour.append(production_mix)
 
+    # if production is the same for at least 3 in the last 48 hours, it's
+    # considered constant. Using round because of floating-point precision.
+    c = Counter([round(sum(v for _, v in e['production'].items()), 3)
+                 for e in production_mix_by_hour])
+    most_common = c.most_common(1)[0]
+    if most_common[1] > 3:
+        raise ValueError(
+            f'Detected constant production in CL-SING. Value {most_common[0]} '
+            f'occured {most_common[1]} times in the last 48 hours.')
```

Comment:
I think you need to remove the initial `f` on the second line of the string continuation for the syntax to work

Reply:
Strings are computed separately then concatenated, so it's right this way.

```python
a, b = 0, 1

(f'this is a test of {a}'
 f' and {b}')
>>> 'this is a test of 0 and 1'

(f'this is a test of {a}'
 ' and {b}')
>>> 'this is a test of 0 and {b}'